### PR TITLE
Fix for Usergrid-22

### PR DESCRIPTION
**Pull request by malakaM**
Link to bug: https://issues.apache.org/jira/browse/USERGRID-22

### CODE REVIEW

**Comment by malakaM** [comment]
Sorry for the confusion with the commits, had two bug fixes and had to revert/figure out how to etc. 
This addresses the null pointer issue, also the same as : https://github.com/usergrid/usergrid/pull/99

File: stack/core/src/main/java/org/apache/usergrid/persistence/Query.java [hunk 1]
```diff
@@ -442,6 +454,11 @@ public UUID getSingleUuidIdentifier() {
         if ( !containsSingleUuidIdentifier() ) {
             return null;
         }
+        //part of fix for the bug USERGRID-22 (to prevent a null pointer exception).
+        if( identifiers.get( 0 ) == null ){
```
Comment (tnine) [inline stack/core/src/main/java/org/apache/usergrid/persistence/Query.java:458]: Does null actually get set into the id list?

Comment (malakaM) [inline stack/core/src/main/java/org/apache/usergrid/persistence/Query.java:458]: Yes @tnine , it does... if this check is not done then the next line ' identifiers.get( 0 ).getUUID()' thrown a null pointer. 
Quite happy that this is getting reviewed :)

**Comment by malakaM** [comment]
@tnine , 
Let me know if any changes needs to be done to the fix.

Thanks, 
Malaka

**Comment by malakaM** [comment]
Thanks for doing the merge @tnine ... :)

**Pull request merged**